packages/x-data-grid/src/components/GridSkeletonLoadingOverlay.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import Skeleton from '@mui/material/Skeleton';
+
+import { styled } from '@mui/system';
```

Comment:
We should import from Material UI here no? Same for all instances, until we deprecate `@mui/material/styles` for `@mui/system`. But until then, we need the theme.

Reply:
@oliviertassinari I wasn't sure because I found other instances in the same directory where `styled` is imported from `@mui/system` e.g. https://github.com/mui/mui-x/blob/master/packages/x-data-grid/src/components/base/GridOverlays.tsx#L3

Reply:
@oliviertassinari I had a similar discussion on Slack last week with quite a different answer: https://mui-org.slack.com/archives/C0170JAM7ML/p1717077013655759

It would be great to align on this topic to avoid back and forth between those two import paths

Reply:
> In the future, everything can be imported from @mui/system (meaning Pigment CSS).

Is there a timeline for this? With the next major?

It might impact a priority [OKR](https://www.notion.so/mui-org/xGrid-Support-using-Data-Grid-with-different-designs-1ad0338b60b0431a960842427e414338?pvs=4#0a4ae7d1d9ca462e8959e8ee26736b91) for the Data Grid where we want to remove all `@mui/material` imports.

Reply:
> Is there a timeline for this? With v6?

I don't know. I'm actually not even sure it's the direction taken by the MUI System team. cc @brijeshb42 and @siriwatknp. At least, in my mind. @mui/material should include nothing else than Base UI components that are styled. I could see Pigment System as where all the theming and specific styling things of Material UI to be hosted or have it under a separate brand like Material UI System.